landlab/components/space/space.py

```diff
@@ -450,91 +447,138 @@ def run_one_step_basic(self, dt=1.0):
+            (
+                self._depo_rate[full_equation * (full_equation * too_thick)]
+                / (1 - self._phi)
+            )
+            - (self._sed_erosion_term[full_equation * too_thick] / (1 - self._phi))
+        ) * dt
+
+        # Finally do the full equation (Eq 32) when not too thick.
+        self._soil__depth[full_equation * (~too_thick)] = self._H_star * np.log(
             (
                 1
                 / (
-                    (self._depo_rate[pos_not_flood] / (1 - self._phi))
-                    / (self._sed_erosion_term[pos_not_flood] / (1 - self._phi))
+                    (self._depo_rate[full_equation * (~too_thick)] / (1 - self._phi))
+                    / (
+                        self._sed_erosion_term[full_equation * (~too_thick)]
+                        / (1 - self._phi)
+                    )
                     - 1
                 )
             )
             * (
                 np.exp(
                     (
-                        self._depo_rate[pos_not_flood] / (1 - self._phi)
-                        - (self._sed_erosion_term[pos_not_flood] / (1 - self._phi))
+                        self._depo_rate[full_equation * (~too_thick)] / (1 - self._phi)
+                        - (
+                            self._sed_erosion_term[full_equation * (~too_thick)]
+                            / (1 - self._phi)
+                        )
                     )
                     * (dt / self._H_star)
                 )
                 * (
                     (
                         (
-                            self._depo_rate[pos_not_flood]
+                            self._depo_rate[full_equation * (~too_thick)]
                             / (1 - self._phi)
-                            / (self._sed_erosion_term[pos_not_flood] / (1 - self._phi))
+                            / (
+                                self._sed_erosion_term[full_equation * (~too_thick)]
+                                / (1 - self._phi)
+                            )
                         )
                         - 1
                     )
-                    * np.exp(self._soil__depth[pos_not_flood] / self._H_star)
+                    * np.exp(H_over_H_star[full_equation * (~too_thick)])
                     + 1
                 )
                 - 1
             )
         )
 
-        # places where slope <= 0 but not flooded:
-        neg_slope_not_flooded = (
-            (self._q > 0) & (~blowup) & (self._slope <= 0) & (~flooded)
-        )
-        self._soil__depth[neg_slope_not_flooded] += (
-            self._depo_rate[neg_slope_not_flooded] / (1 - self._phi) * dt
-        )
+        # Equation 12 gives dRdt, and Equation 36 gives R. However, we don't
+        # include dH/dt within timestep in integrating for R.
+        # This matters when we are starting with very little soil and increasing.
+
+        # to do this right I think we need a numerical integral for
```

Comment:
Good at least to have this on the radar screen, thanks for leaving the breadcrumbs!